packages/core/src/index.ts

```diff
@@ -4,25 +4,26 @@ import axios from "axios";
 interface Config {
   apiKey: string;
   botId: string;
+  contactId: string;
+  debug?: boolean;
   journeyId?: string;
-  voice?: string;
   language?: string;
-  contactId: string;
+  voice?: string;
 }
 
 // The journey manager object
 export interface VoiceCompass {
-  updateStep: (data: StepData) => void;
+  updateStep: <T>(data: StepData<T>) => void;
   trackDomAnnotations: () => void;
   stopTrackingDomAnnotations: () => void;
 }
 
-interface StepData {
+interface StepData<T> {
   stepId: string;
-  journeyId?: string;
+  journeyId?: Config["journeyId"];
   end?: boolean;
   escalate?: boolean;
-  payload?: object;
+  payload?: T;
```

Comment:
i think generics is more appropriate than an object?

Reply:
I think having a generic here would mislead the user about what this payload is - it really can be anything at this point by design. When you use the SDK in a project, you could create a custom payload interface and have the payloads you generate conform to it.

Maybe `Record<string, any>` is a better type here but let's leave that until a next release (feel free to open an issue).